plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/TikaImpl.java

```diff
@@ -161,8 +162,12 @@ static PermissionCollection getRestrictedPermissions() {
         perms.add(new ReflectPermission("suppressAccessChecks"));
         // xmlbeans, use by POI, needs to get the context classloader
         perms.add(new RuntimePermission("getClassLoader"));
-        // ZipFile needs accessDeclaredMembers on Java 10
-        perms.add(new RuntimePermission("accessDeclaredMembers"));
+        // ZipFile needs accessDeclaredMembers on JDK 10; cf. https://bugs.openjdk.java.net/browse/JDK-8187485
+        if (JavaVersion.current().compareTo(JavaVersion.parse("10")) >= 0) {
```

Comment:
Out of curiosity, we cannot do a check with the similar intention in the policy file itself, no? To me it looks like this part of the code is only used for testing.

Reply:
This is not only used in tests, trace from [`AttachmentProcessor#execute`](https://github.com/elastic/elasticsearch/blob/43ff38c5dabcd862e260db28efd951d077cdf777/plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java#L88) to see that the restricted permissions are used for parsing. We can not do a check in the policy file indeed, that's why I leave a comment to say also remove this from the policy file when we can.

Reply:
Right, the method comment ("// only package private for testing!") put me on the wrong track. Thanks for the clarification. Maybe that comment should be deleted then.

Reply:
The reason the method is package private as opposed to private is only for testing, I think the comment is correct?

Reply:
No, it cannot be private, since it's used by `AttachmentProcessor` in the same package. That was the part that I was missing. I think its misleading.

Reply:
You're right. Let us remove the comment.

Reply:
I can push the change if you don't have it ready yet.

Reply:
Go ahead, thank you.